hardware/Netatmo.cpp

```diff
@@ -504,8 +674,167 @@ uint64_t CNetatmo::UpdateValueInt(int HardwareID, const char* ID, unsigned char
                 Log(LOG_NORM, szLogString);
         }
         m_mainworker.sOnDeviceReceived(m_HwdID, DeviceRowIdx, devname, nullptr);
-        m_notifications.CheckAndHandleNotification(DeviceRowIdx, m_HwdID, ID, devname, unit, devType, subType, nValue, sValue);
+        m_notifications.CheckAndHandleNotification(DeviceRowIdx, m_HwdID, sDeviceID.c_str(), devname, unit, devType, subType, nValue, sValue);
         m_mainworker.CheckSceneCode(DeviceRowIdx, devType, subType, nValue, sValue, "MQTT Auto");
+
+	if (m_bMigrationFlag) {
+//		Log(LOG_STATUS, "UpdateValueInt: DeviceRowIdx=%d, HardwareID=%d (%X), ID=%s (%s), unit=%d, devType=%d, subType=%d, signallevel=%d, batterylevel=%d, nValue=%d, sValue=%s, devname=%s, bUseOnOffAction=%d, user=%s", 
+//			(int)DeviceRowIdx,
+//			HardwareID, 
+//			HardwareID, 
+//			deviceID, 
+//			sDeviceID.c_str(), 
+//			unit, 
+//			devType, 
+//			subType, 
+//			signallevel, 
+//			batterylevel, 
+//			nValue, 
+//			sValue, 
+//			devname.c_str(), 
+//			bUseOnOffAction, 
+//			user.c_str());
+//
+//		Log(LOG_STATUS,"Record added/updated with DeviceRowIdx = %d", (int)DeviceRowIdx);
+
+//		Log(LOG_STATUS, "UpdateValueInt: DeviceRowIdx=%d, ID=%s (%s), unit=%d, devType=%d, subType=%d, devname=%s", 
+//			(int)DeviceRowIdx,
+//			deviceID, 
+//			sDeviceID.c_str(), 
+//			unit, 
+//			devType, 
+//			subType, 
+//			devname.c_str());
+
+		std::vector<std::vector<std::string> > result;
+		std::string searchString = sDeviceID.substr(sDeviceID.length()-2,2);
+
+		result = m_sql.safe_query(
```

Comment:
Please keep the MySQL functions case sensitive, like most of them are in CAPITAL letters, see the documentation

https://www.w3schools.com/sql/func_mysql_substr.asp

Also check this throughout this PR

Reply:
OK (I suppose you mean SUBSTR)

Reply:
correct and also some date/time functions, most things are case sensitive, like SELECT instead of Select for instance (not that you did this... )

Reply:
The example above is not a SQL syntax, but C++, so it needs to be in lower case in this case. However, I modified all other cases where it was part of a SQL statement.